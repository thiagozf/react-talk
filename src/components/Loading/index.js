// @flow

import React, { Component } from 'react';
import ReactLoading from 'react-loading';

import { colors } from 'components/constants';

import styled from 'styled-components';

const Wrapper = styled.div`
  width: 200px;
  margin: 0 auto;
`;

type Props = {};

export default class Loading extends Component<Props>  {
  render() {
    return (
      <Wrapper>
        <ReactLoading styles={{ margin: '0 auto' }} type="bubbles" color={colors.primary} height={50} width={200} />
      </Wrapper>
    );
  }
}
