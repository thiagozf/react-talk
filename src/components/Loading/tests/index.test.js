import React from 'react';
import { shallow } from 'enzyme';
import ReactLoading from 'react-loading';

import Loading from '../';

describe('Loading tests', () => {
  it('should render', () => {
    const loading = shallow(<Loading />);
    expect(loading).toBeDefined();
  });

  it('should present bubbles', () => {
    const loading = shallow(<Loading />);
    const reactLoading = loading.find(ReactLoading);
    expect(reactLoading).toBeDefined();
    expect(reactLoading.props().type).toBe('bubbles');
  });

  it('should match UI defs', () => {
    const loading = shallow(<Loading />);
    expect(loading).toMatchSnapshot();
  });
});
