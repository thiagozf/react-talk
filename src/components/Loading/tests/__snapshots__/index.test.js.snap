// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading tests should match UI defs 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Loading
        color="#303c76"
        delay={1000}
        height={50}
        styles={
                Object {
                        "margin": "0 auto",
                      }
        }
        type="bubbles"
        width={200}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "color": "#303c76",
        "delay": 1000,
        "height": 50,
        "styles": Object {
          "margin": "0 auto",
        },
        "type": "bubbles",
        "width": 200,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Loading
          color="#303c76"
          delay={1000}
          height={50}
          styles={
                    Object {
                              "margin": "0 auto",
                            }
          }
          type="bubbles"
          width={200}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "color": "#303c76",
          "delay": 1000,
          "height": 50,
          "styles": Object {
            "margin": "0 auto",
          },
          "type": "bubbles",
          "width": 200,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
